diff -u3 scintilla/gtk/PlatGTK.orig.cxx scintilla/PlatGTK.cxx
--- scintilla/gtk/PlatGTK.orig.cxx	2011-09-16 11:13:20.000000000 +0200
+++ scintilla/PlatGTK.cxx	2011-09-17 10:33:32.458041001 +0200
@@ -719,6 +719,16 @@
 	}
 	return new FontHandle(newid);
 #else
+	// Try to use pango font even if it's not specified
+	PangoFontDescription *pfd = pango_font_description_new();
+	if (pfd) {
+		pango_font_description_set_family(pfd, fontName);
+		pango_font_description_set_size(pfd, size * PANGO_SCALE);
+		pango_font_description_set_weight(pfd, bold ? PANGO_WEIGHT_BOLD : PANGO_WEIGHT_NORMAL);
+		pango_font_description_set_style(pfd, italic ? PANGO_STYLE_ITALIC : PANGO_STYLE_NORMAL);
+		return new FontHandle(pfd, characterSet);
+	}
+
 	return new FontHandle();
 #endif
 }
