--- scintilla/scintilla/include/Scintilla.h	2009-11-02 05:51:51.000000000 +0100
+++ scintilla/include/Scintilla.h	2010-01-31 19:42:12.000000000 +0100
@@ -36,6 +36,11 @@
 #if defined(_WIN32)
 #include <basetsd.h>
 #endif
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+typedef uintptr_t uptr_t;
+typedef intptr_t sptr_t;
+#else
 #ifdef MAXULONG_PTR
 typedef ULONG_PTR uptr_t;
 typedef LONG_PTR sptr_t;
@@ -43,6 +48,7 @@
 typedef unsigned long uptr_t;
 typedef long sptr_t;
 #endif
+#endif
 
 typedef sptr_t (*SciFnDirect)(sptr_t ptr, unsigned int iMessage, uptr_t wParam, sptr_t lParam);
 
