diff -u3 include/Scintilla.orig.h include/Scintilla.h
--- include/Scintilla.orig.h	2011-08-01 04:49:04.000000000 +0200
+++ include/Scintilla.h	2011-09-17 10:54:12.743040993 +0200
@@ -32,6 +32,11 @@
 #if defined(_WIN32)
 #include <basetsd.h>
 #endif
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+typedef uintptr_t uptr_t;
+typedef intptr_t sptr_t;
+#else
 #ifdef MAXULONG_PTR
 typedef ULONG_PTR uptr_t;
 typedef LONG_PTR sptr_t;
@@ -39,6 +44,7 @@
 typedef unsigned long uptr_t;
 typedef long sptr_t;
 #endif
+#endif
 
 typedef sptr_t (*SciFnDirect)(sptr_t ptr, unsigned int iMessage, uptr_t wParam, sptr_t lParam);
 
